pkg/exemplars/exemplars.go

```diff
@@ -0,0 +1,145 @@
+package exemplars
+
+import (
+	"context"
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/thanos-io/thanos/pkg/exemplars/exemplarspb"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+)
+
+var _ UnaryClient = &GRPCClient{}
+
+// UnaryClient is gRPC exemplarspb.Exemplars client which expands streaming exemplars API. Useful for consumers that does not
+// support streaming.
+type UnaryClient interface {
+	Exemplars(ctx context.Context, req *exemplarspb.ExemplarsRequest) ([]*exemplarspb.ExemplarData, storage.Warnings, error)
+}
+
+// GRPCClient allows to retrieve exemplars from local gRPC streaming server implementation.
+// TODO(bwplotka): Switch to native gRPC transparent client->server adapter once available.
+type GRPCClient struct {
+	proxy exemplarspb.ExemplarsServer
+
+	replicaLabels map[string]struct{}
+}
+
+type exemplarsServer struct {
+	// This field just exist to pseudo-implement the unused methods of the interface.
+	exemplarspb.Exemplars_ExemplarsServer
+	ctx context.Context
+
+	warnings []error
+	data     []*exemplarspb.ExemplarData
+}
```

Comment:
Let's add two additional methods for it https://github.com/yeya24/thanos/commit/1b6abfcb35e2a7153a9dabda921926a7f72b8c7f#diff-d09c4014e056aee5c67a76aca788de273aecb440edefc8a3b040bc77e27f2c38R39-R55, otherwise it will panic.